docs/proposals/config-dump.md

```diff
@@ -0,0 +1,248 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should refer to the docs that go with that version.
+
+Documentation for other releases can be found at [releases.K8s.io](http://releases.K8s.io).
+</strong>
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+
+# Overview
+
+Here we describe a proposal for retrieving configurations (configs) that govern the behavior of any one of the Kubernetes (K8s) components. By components we mean any of the following: apiserver, scheduler, control-manager, and kubelet. We expect that these configurations be for the most part static through the life of the component but fully understand that changes from specs may affect these. Consequently, any retrieval of the configurations is to be considered true and valid at the time of retrieval and we will assume that the user is well aware of any transient behavior.   
+
+We propose several important changes:
+* new kubectl command: 
+    * `kubectl get componentconfigs <componentName> (alias compcs) --id <componentId>`
```

Comment:
We talked about this some this morning. As we move to component config and away from flags, the component config is the natural thing to grab.  There are still going to be some things that are note captured in the component config (probably etcd info?) but it should be isolated.

However, we can't yet assume that everyone is selfhosting and grabbing the componentconfig from configmaps.  At the community meeting today we saw an example (RackN/Digital Rebar) of tooling that chooses not to use self-hosting.

An alternative that was brought up would be to support a "give me your config" API on all of our services.  We'd lock this down from a security point of view.  A tool could then iterate all services in the `kube-system` namespace and hit the endpoints for those services at a new API path and collect the results.  With external services this could work even if components are not self hosted.  (as a bonus we could specify an annotation of where to find this endpoint if it is not in the default place).

I agree, however, that we should avoid adding a new API here if we can help it at all.